hudi-common/src/main/java/org/apache/hudi/metadata/MetadataPartitionType.java

```diff
@@ -332,13 +332,13 @@ public String getIndexNameWithoutPrefix(HoodieIndexDefinition indexDefinition) {
    */
   public static boolean isGenericIndex(String metadataPartitionPath) {
     return metadataPartitionPath.startsWith(SECONDARY_INDEX.getPartitionPath())
-        || metadataPartitionPath.startsWith(FUNCTIONAL_INDEX.getPartitionPath());
+        || metadataPartitionPath.startsWith(EXPRESSION_INDEX.getPartitionPath());
```

Comment:
can we reuse `MetadataPartitionType.fromParttionPath`?

Reply:
Addressed